Tests/IntegrationTests/Components Tests/ACH Direct Debit/ACHDirectDebitComponentTests.swift

```diff
@@ -304,4 +304,65 @@ class ACHDirectDebitComponentTests: XCTestCase {
         let infoType = analyticsProviderMock.infos.first?.type
         XCTAssertEqual(infoType, .rendered)
     }
+
+    func testSubmitWithDefaultSubmitHiddenShouldCallPaymentDelegateDidSubmit() throws {
+        // Given
+        let paymentMethod = ACHDirectDebitPaymentMethod(type: .achDirectDebit, name: "Test name")
+        let configuration = ACHDirectDebitComponent.Configuration(showsSubmitButton: false,
+                                                                  showsBillingAddress: false)
+        let sut = ACHDirectDebitComponent(paymentMethod: paymentMethod,
+                                          context: context,
+                                          configuration: configuration,
+                                          publicKeyProvider: PublicKeyProviderMock())
+
+        setupRootViewController(sut.viewController)
+
+        let expectation = XCTestExpectation(description: "Dummy Expectation")
+
+        let paymentDelegateMock = PaymentComponentDelegateMock()
+        sut.delegate = paymentDelegateMock
+        paymentDelegateMock.onDidSubmit = { _, component in
+            XCTAssertTrue(component === sut)
+            expectation.fulfill()
+        }
+
+        let nameItemView: FormTextItemView<FormTextInputItem> = try XCTUnwrap(sut.viewController.view.findView(with: "AdyenComponents.ACHDirectDebitComponent.holderNameItem"))
+        let accountNumberItemView: FormTextItemView<FormTextInputItem> = try XCTUnwrap(sut.viewController.view.findView(with: "AdyenComponents.ACHDirectDebitComponent.bankAccountNumberItem"))
+        let routingNumberItemView: FormTextItemView<FormTextInputItem> = try XCTUnwrap(sut.viewController.view.findView(with: "AdyenComponents.ACHDirectDebitComponent.bankRoutingNumberItem"))
+
+        self.populate(textItemView: nameItemView, with: "test")
+        self.populate(textItemView: accountNumberItemView, with: "123456789")
+        self.populate(textItemView: routingNumberItemView, with: "121000358")
+
+        // When
+        sut.submit()
+
+        // Then
+        wait(for: [expectation], timeout: 100)
```

Comment:
This should not be more than 10 - it does not slow down the tests but if it takes longer there is usually something else wrong - we had cases where CI was really slow but ideally the tests then just fail and don't block the CI for over a minute

Reply:
Mistake from my side, I intended to use 10 seconds, not 100. Fixed